pilot/pkg/config/monitor/monitor.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"reflect"
+	"strings"
+	"time"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+// Monitor will poll a config function in order to update a ConfigStore as
+// changes are found.
+type Monitor struct {
```

Comment:
I assume a file notifier will be used in future ?

Reply:
The `filesnapshotter` is actually already being used in the code (its in bootstrap). This monitor abstraction now allows us to introduce a `copilotsnapshotter` but that's in another PR>